crates/compiler/parse/src/expr.rs

```diff
@@ -1563,11 +1563,24 @@ fn parse_after_binop<'a>(
     mut expr_state: ExprState<'a>,
     loc_op: Loc<BinOp>,
 ) -> ParseResult<'a, Expr<'a>, EExpr<'a>> {
-    match loc_possibly_negative_or_negated_term(options).parse(
-        arena,
-        state.clone(),
-        call_min_indent,
-    ) {
+    // The `dbg` keyword with no args is only allowed after a `|>`
+    let is_after_pizza = matches!(
+        loc_op,
+        Loc {
+            value: BinOp::Pizza,
+            ..
+        }
+    );
+
+    let parser = one_of!(
+        loc(specialize_err(
+            EExpr::Dbg,
+            dbg_after_pizza_help(options, is_after_pizza)
```

Comment:
This is kind of a goofy hack. We only want dbg to be an option when the previous bin op was `|>`, so I tried to implement it as
```
    let dbg_parser = if is_after_pizza {
        dbg_after_pizza_help(options)
    } else {
        parser::fail(EExpect::IndentCondition)
    };

    let parser = one_of!(
        loc(specialize_err(EExpr::Dbg, dbg_parser)),
        loc_possibly_negative_or_negated_term(options),
    );

```

Which gives the error
```
Diagnostics:
1. `if` and `else` have incompatible types
   expected opaque type `impl parser::Parser<'_, ast::Expr<'_>, EExpect<'_>>`
      found opaque type `impl parser::Parser<'_, _, EExpect<'_>>`
   distinct uses of `impl Trait` result in different opaque types [E0308]
```
I'm not sure how to approach this, so instead I passed `is_after_pizza` into `dbg_after_pizza_help` and put the short-circuit logic in there.

Reply:
@joshuawarner32 -- can you have a look at this please?

Reply:
Yeahhhhhh. Your initial attempt is impossible-ish to work because of rust's type system.

Reply:
I cut this from the PR, but parsing dbg as a keyword should avoid this silliness.